test/k8sT/Policies.go

```diff
@@ -478,68 +471,59 @@ var _ = Describe("K8sValidatedPolicyTest", func() {
 
 		It("Denies traffic with cnp default-deny ingress policy", func() {
 
-			res := kubectl.ExecPodCmd(
-				helpers.DefaultNamespace, appPods[helpers.App2],
-				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			res.ExpectSuccess("%q cannot curl clusterIP %q", appPods[helpers.App2], clusterIP)
-
-			res = kubectl.ExecPodCmd(
-				helpers.DefaultNamespace, appPods[helpers.App3],
-				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			res.ExpectSuccess("%q cannot curl to %q", appPods[helpers.App3], clusterIP)
-
 			By("Installing cnp ingress default-deny")
 
 			_, err := kubectl.CiliumPolicyAction(
 				helpers.KubeSystemNamespace, cnpDenyIngress, helpers.KubectlApply, 300)
 			Expect(err).Should(BeNil(),
 				"L3 deny-ingress Policy cannot be applied in %q namespace", helpers.DefaultNamespace)
 
+			Expect(kubectl.WaitCEPReady()).To(BeNil(), "Cep is not ready in a timeout")
+			for _, app := range []string{helpers.App1, helpers.App2, helpers.App3} {
+				pod := appPods[app]
+				cep := kubectl.CepGet(helpers.DefaultNamespace, pod)
+				Expect(cep).NotTo(BeNil(), "Endpoint %q does not exits", pod)
+
+				Expect(cep.Status.Policy.Realized.PolicyEnabled).To(
+					Equal(models.EndpointPolicyEnabledIngress),
+					"Policy status does not match for endpoint %s", pod)
+			}
+
 			By("Testing connectivity with ingress default-deny policy loaded")
 
-			res = kubectl.ExecPodCmd(
+			res := kubectl.ExecPodCmd(
 				helpers.DefaultNamespace, appPods[helpers.App2],
 				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
 			res.ExpectFail("Ingress connectivity should be denied by policy")
 
+			By("Testing egress connnectivity works correctly")
 			res = kubectl.ExecPodCmd(
-				helpers.DefaultNamespace, appPods[helpers.App3],
-				helpers.CurlFail(fmt.Sprintf("http://%s/public", clusterIP)))
-			res.ExpectFail("Ingress connectivity should be denied by policy")
+				helpers.DefaultNamespace, appPods[helpers.App2],
+				helpers.Ping("8.8.8.8"))
```

Comment:
Why the move from curl to ping?

Reply:
Because we tested from APP2/APP3 to connect to clusterIP (app1-service) but we don't test any egress to world connectivity.